### ISSUE TITLE
BlueWallet Removed from F-Droid

### ISSUE THREAD

**Issue by jacobranson** [BlueWallet/BlueWallet#5047]
As per the recent discussion in #233, (starting [here](https://github.com/BlueWallet/BlueWallet/issues/233#issuecomment-1081768103)) the F-Droid builds of BlueWallet have been removed due to the react-native-camera dependency having non-FOSS upstream dependencies. I thought it might be pertinent to open a new issue to track the regression, although the previous issue could be re-opened instead, if the devs prefer.

**Comment by tomichec** [comment]
It's a real loss that the app is no longer in the F-Droid repository. 

Is there any progress on getting it there again? @pointbiz @erfanoabdi @Overtorment 

I would be happy to help.

How to find out if the suggested [potential replacement](https://github.com/BlueWallet/BlueWallet/issues/233#issuecomment-1099514212) is suitable (without non-free dependencies)?

**Comment by grigio** [comment]
it's a pity that this is still an issue

**Comment by 48-mq** [comment]
@Overtorment @marcosrdz @limpbrains Would you please let us know what you intend to do about this?

Is the intention to keep proprietary & uninspected code inside of the app? If so, users need to know so we can start switching to a different wallet.

**Comment by marcosrdz** [comment]
> @Overtorment @marcosrdz @limpbrains Would you please let us know what you intend to do about this?
> 
> 
> 
> Is the intention to keep proprietary & uninspected code inside of the app? If so, users need to know so we can start switching to a different wallet.

The whole app is open source.

**Comment by 48-mq** [comment]
@marcosrdz Please elaborate. 

Is com.google.android.gms:play-services-mlkit-barcode-scanning not a dependency anymore? Has react-native-camera finally been replaced?

This is an important issue, since it affects the inspectability of the code.

**Comment by marcosrdz** [comment]
> @marcosrdz Please elaborate. 
> 
> 
> 
> Is com.google.android.gms:play-services-mlkit-barcode-scanning not a dependency anymore? Has react-native-camera finally been replaced?
> 
> 
> 
> This is an important issue, since it affects the inspectability of the code.

React native camera was replaced a long time ago as that project was abandoned

**Comment by 48-mq** [comment]
Amazing! Thank you for the confirmation @marcosrdz 

Now let's see if this can be put back in the F-Droid repo.

**Comment by Overtorment** [comment]
Tbf new camera dep i think also has Google dep. Creating degoogled qr scanner is a bit non trivial, but has been on my radar for a long time, tho not a high priority

**Comment by 48-mq** [comment]
Google dependency is okay, as long as all the code is freely available for inspection. This may be how the other wallets on F-Droid are doing it (I'm only guessing).

Let's see what the F-Droid dev says on #233 ...

**Comment by licaon-kter** [comment]
> Google dependency is okay

lol wut? still depends on proprietary Google libs, https://github.com/teslamotors/react-native-camera-kit/blob/master/android/build.gradle#L47

**Comment by 48-mq** [comment]
> > @marcosrdz Please elaborate.
> > Is com.google.android.gms:play-services-mlkit-barcode-scanning not a dependency anymore? Has react-native-camera finally been replaced?
> > This is an important issue, since it affects the inspectability of the code.
> 
> React native camera was replaced a long time ago as that project was abandoned

@marcosrdz I'm still seeing references to only react-native-camera in the code. What is the replacement that you mention was put in place a long time ago?

**Comment by 48-mq** [comment]
@Overtorment Please feel free to chime in here

**Comment by marcosrdz** [comment]
> > > @marcosrdz Please elaborate.
> 
> > > Is com.google.android.gms:play-services-mlkit-barcode-scanning not a dependency anymore? Has react-native-camera finally been replaced?
> 
> > > This is an important issue, since it affects the inspectability of the code.
> 
> > 
> 
> > React native camera was replaced a long time ago as that project was abandoned
> 
> 
> 
> @marcosrdz I'm still seeing references to only react-native-camera in the code. What is the replacement that you mention was put in place a long time ago?

react-native-camera and react-native-camera-kit are two different projects

**Comment by 48-mq** [comment]
So essentially we're in the same place as before? It cannot be proved, for instance, that com.google.mlkit:barcode-scanning does not have malicious code in it, or something that will result in loss of funds.

**Comment by alex27riva** [comment]
Hi everyone, any update on this?

**Comment by tomichec** [comment]
I don't think there is any update. BlueWallet app is still not fully free software (and open source) and therefore can not be included to f-droid.

This is because the application 'react-native-camera-kit' that is being used depends on [the proprietary ML KIT ][1] providing ['com.google.mlkit:barcode-scanning'][2].

I just checked that the suggested [potential replacement ][3] the [react-native-vision-camera ][4] suffers the same issue (dependency on [ML KIT][5]).

Does anyone know what other qr scanner library might potentially solve the issue? What are the requirements of such library? I have done a quick search, but only found other free software apps with qr-scanner written in Java or Kotlin. Would that  be suitable, or does the library need to be in JavaScript?


[1]: https://opensource.stackexchange.com/a/14675

[2]: https://github.com/teslamotors/react-native-camera-kit/blob/29e58736337b18345e88e5d3803a19f7e34680e1/android/build.gradle#L67

[3]: https://github.com/BlueWallet/BlueWallet/issues/233#issuecomment-1099514212

[4]: https://github.com/mrousavy/react-native-vision-camera

[5]: https://github.com/mrousavy/react-native-vision-camera/blob/8e919c8ba574cf766463666b82d7ad2180855f1f/package/android/build.gradle#L204

**Comment by theborakompanioni** [comment]
> I don't think there is any update. BlueWallet app is still not fully free software (and open source) and therefore can not be included to f-droid.

Would it be possible to run [fdroidserver](https://github.com/f-droid/fdroidserver) and [host an own instance](https://f-droid.org/docs/Installing_the_Server_and_Repo_Tools/) at `fdroid.bluewallet.io` or something like zeus does (with https://fdroid.zeusln.app/)? Or maybe even ask @kaloudis if bluewallet can be hosted on their fdroid instance? I think that would be a pretty neat workaround and a nice collaboration.

**Comment by Giszmo** [comment]
@theborakompanioni an increasingly popular option is ZapStore. Zeus also is there: https://zapstore.dev/apps/naddr1qvzqqqr7pvpzqdxj75n57x2clnfvkfrrm2l2mhu2y8uy4njzg8dg3qprhuzueqy4qq8kzurs9eax2atnd3hzu7n9w4essgljvy
ZapStore is built on nostr with the goal of being completely decentralized and WoT based and thus not as restrictive as FDroid.

**Comment by licaon-kter** [comment]
@Giszmo "not as restrictive" means "accepts proprietary apps"?

**Comment by tomichec** [comment]
Exactly, and those "restrictions" I see as the main advantage of getting an app from the official F-droid's repository.

It's would not be a problem to get an up from an alternative source, but I tend to trust the software more when I know there is someone independent to check whether the program is really freedom respecting and does not rely on proprietary software or infrastructure.

I think that the dependency on proprietary libraries (which in the case of BlueWallet is the com.google.mlkit:barcode-scanning) should really be treated as a bug, especially in the case when the application is advertised as "open source".

**Comment by tomichec** [comment]
I saw ZeusLN people are [suggesting](zeusissue) to replace the proprietary ML KIT with free software stand-alone QR library: https://github.com/limpbrains/qr

[zeusissue]: https://github.com/ZeusLN/zeus/issues/3493

**Comment by limpbrains** [comment]
> I saw ZeusLN people are [suggesting](zeusissue) to replace the proprietary ML KIT with free software stand-alone QR library: https://github.com/limpbrains/qr

in BlueWallet we have recently switched to https://github.com/limpbrains/react-native-camera-kit-no-google
which is using https://github.com/limpbrains/qr under the hood

**Comment by licaon-kter** [comment]
@limpbrains so the app is ready to be reactivated in F-Droid?

**Comment by limpbrains** [comment]
Technically, the master branch is ready, but please give it a bit more time.
We haven’t released a public version of BW with this new camera yet, and we need to make sure it works for everyone.

**Comment by licaon-kter** [comment]
thanks, ping us when ready
